### ISSUE TITLE
CMake.py not detecting multi-configuration builds correctly

### ISSUE THREAD

**Issue by 16k** [conan-io/conan#700]
This is to resolve a problem when using Visual Studio while setting CONAN_CMAKE_GENERATOR=Ninja. This caused the debug version to always be built.

The function is_multi_configuration() is basing its decision on self._settings.compiler. If instead it called generator() then we could return true for the cmake generators that produce multi-configurations.

**Comment by memsharded** [comment]
I am not sure if I understood your setup. If you are setting the cmake generator for Ninja, then, you are not building with Visual Studio, those are different CMake generators, aren't they?

**Comment by 16k** [comment]
You can use different compilers with the Ninja generator.

I'm building on windows and it seems that CMake usually pulls out the first compiler it finds in the path or it uses the CC environment variable. I've been selecting the compiler used simply by setting up specific environments with only one or the other compiler available. If both are available in the path you can select which one is used with CMAKE_C_COMPILER.

For example, with both gcc and cl in the path:
```
C:\Users\williami\Test\zlib-1.2.8\build>gcc
gcc: fatal error: no input files
compilation terminated.

C:\Users\williami\Test\zlib-1.2.8\build>cl
Microsoft (R) C/C++ Optimizing Compiler Version 19.00.24215.1 for x86
Copyright (C) Microsoft Corporation.  All rights reserved.

usage: cl [ option... ] filename... [ /link linkoption... ]

C:\Users\williami\Test\zlib-1.2.8\build>cmake -G "Ninja" -DCMAKE_C_COMPILER=cl -DCMAKE_BUILD_TYPE=Release ..
-- The C compiler identification is MSVC 19.0.24215.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - failed
-- Looking for fseeko
-- Looking for fseeko - not found
-- Looking for unistd.h
-- Looking for unistd.h - not found
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/williami/Test/zlib-1.2.8/build

C:\Users\williami\Test\zlib-1.2.8\build>cmake -G "Ninja" -DCMAKE_C_COMPILER=gcc -DCMAKE_BUILD_TYPE=Release ..
-- The C compiler identification is GNU 4.9.2
-- Check for working C compiler: C:/C/mingw-installer/0.1/plex/stable/package/8452cab1d9686aac55e5b8411dbfa277c73e40ba/bin/gcc.exe
-- Check for working C compiler: C:/C/mingw-installer/0.1/plex/stable/package/8452cab1d9686aac55e5b8411dbfa277c73e40ba/bin/gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - done
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/williami/Test/zlib-1.2.8/build
```

What I'm trying to do is to generate either gcc or Visual C++ built binaries simply by changing my conan profile.

**Comment by memsharded** [comment]
Understood, submitted PR #702, was easy to implement, just what you said, next time go ahead and submit a PR yourself, and get the credit :)

Thanks for your feedback!

**Comment by 16k** [comment]
Thanks very much! 😄

**Comment by memsharded** [comment]
Released in 0.16!
